pipeline/ruleset.go

```diff
@@ -122,170 +125,170 @@ func (r *Rules) Empty() bool {
 // ruletypes from the rules match the provided ruledata. For
 // both operators, when none of the ruletypes from the rules
 // match the provided ruledata, the function returns false.
-//
-//nolint:gocyclo // accepting complexity in this case
-func (r *Rules) Match(from *RuleData, matcher, op string) bool {
-	// set defaults
-	status := true
+func (r *Rules) Match(from *RuleData, matcher, op string) (bool, error) {
+	var err error
 
 	// if the path ruletype is provided
 	if len(from.Path) > 0 {
 		// if the "or" operator is provided in the ruleset
 		if strings.EqualFold(op, constants.OperatorOr) {
-			// override the default to the "or"
-			if len(from.Status) != 0 {
-				status = r.Status.MatchOr(from.Status, matcher)
-			}
-
 			// iterate through each path in the ruletype
 			for _, p := range from.Path {
+				matches, err := matches(r, from, matcher, p, constants.OperatorOr)
+				if err != nil {
+					return false, err
+				}
+
 				// return true if any ruletype matches the ruledata
-				if r.Branch.MatchOr(from.Branch, matcher) ||
-					r.Comment.MatchOr(from.Comment, matcher) ||
-					r.Event.MatchOr(from.Event, matcher) ||
-					r.Path.MatchOr(p, matcher) ||
-					r.Repo.MatchOr(from.Repo, matcher) ||
-					status ||
-					r.Tag.MatchOr(from.Tag, matcher) ||
-					r.Target.MatchOr(from.Target, matcher) {
-					return true
+				if matches {
+					return true, nil
 				}
```

Comment:
reduce to `return matches(r, from, matcher, p, constants.OperatorOr)`?

Reply:
Same idea with the looping situation as above ^^